src/ImageSharp/Image/IReadonlyPixelAccessor{TColor}.cs

```diff
@@ -0,0 +1,50 @@
+﻿// <copyright file="IReadonlyPixelAccessor{TColor}.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp
+{
+    using System;
+    using System.Diagnostics;
+    using System.Runtime.CompilerServices;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// Provides per-pixel readonly access to generic <see cref="Image{TColor}"/> pixels.
+    /// </summary>
+    /// <typeparam name="TColor">The pixel format.</typeparam>
+    public interface IReadonlyPixelAccessor<TColor>
```

Comment:
My logic with the interface was I explicitly didn't want the brush to have write access to the pixels, the only other way I can think to do it would be with a very thin struct wrapper around `PixelAccessor` with aggressive inlining hints.